pkg/cache/scheduler_callback.go

```diff
@@ -60,8 +60,15 @@ func (callback *AsyncRMCallback) UpdateAllocation(response *si.AllocationRespons
 			return err
 		}
 		if app := callback.context.GetApplication(alloc.ApplicationID); app != nil {
-			ev := NewAllocateTaskEvent(app.GetApplicationID(), alloc.AllocationKey, alloc.AllocationID, alloc.NodeID)
-			dispatcher.Dispatch(ev)
+			if task := callback.context.getTask(app.GetApplicationID(), alloc.AllocationKey); task != nil {
+				if utils.IsAssignedPod(task.GetTaskPod()) {
```

Comment:
Should we exclude the `failed` pod?

Reply:
Failed pods are handled via rejections and won't be here.